src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,99 @@
+     *     3: >
+     *        Interruptive Modals MUST contain a Default Button continuing the action that initiated the Modal (e.g. Delete the Item) on the left side of the footer of the modal and a button canceling the action on the right side of the footer as Default Button.
+     *     4: >
+     *        The Cancel Button in the footer and the Close Button in the header MUST NOT perform any additional action than closing the confirmation screen.
+     * ---
+     * @param string $title
+     * @param Component\Component $content
+     * @return \ILIAS\UI\Component\Modal\Interruptive
+     */
+    public function interruptive($title, Component\Component $content);
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     Round-Trip Modals are to be used if the context would be lost by performing this action otherwise. Round-Trip Modals accommodate sub-workflows within an overriding workflow. The Round-Trip Modal ensures that the users does not leave the trajectory of the overriding workflow. This is typically the case if an ILIAS service is called for while in working in an object.
+     *   composition: >
+     *     Round-Trip Modals are completed by a small well defined sequence of 2 steps max. that might be displayed on a sequence of different modals connected through some “next” Button.
+     *   effect: >
+     *     Round-Trip Modals perform sub-workflow involving some kind of user input. Sub-workflow is completed and user is returned to starting point allowing for continuing the overriding workflow.
+     *
+     * rules:
+     *   usage:
+     *     1: >
+     *       Round-Trip Modals MUST contain at least two buttons at the bottom of the modals: a button to cancel (right) the workflow and a button to finish or reach the next step in the workflow (left).
+     *     2: >
+     *       Round-Trip Modals SHOULD be used, if the user would lose the context otherwise. If the action can be performed within the same context (e.g. add a post in a forum, edit a wiki page), a Round-Trip Modal MUST NOT be used.
+     *     3: >
+     *       When the workflow is completed, Round-Trip Modals SHOULD show the same view as was displayed when initiating the Modal.
+     *     4: >
+     *       Round-Trip Modals SHOULD NOT be used to add new items of any kind since adding item is a linear workflow redirecting to the newly added item setting- or content-tab.
+     *     5: >
+     *       Round-Trip Modals SHOULD NOT be used to perform complex workflows.
+     *
+     * ---
+     * @param string $title
+     * @param Component\Component $content
+     * @return \ILIAS\UI\Component\Modal\RoundTrip
+     */
+    public function roundtrip($title, Component\Component $content);
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The Lightbox Modal displays a full of data such as documents, images or a movie inside a Lightbox.
```

Comment:
This was already weird in the FR. What we tried to say was complete (or maybe full) image of data such as ...

This wording could probably improved.